base/arraymath.jl

```diff
@@ -472,12 +469,16 @@ for (f, f!, fp, op) = ((:cumsum, :cumsum!, :cumsum_pairwise!, :+),
     @eval function ($f!)(result::AbstractVector, v::AbstractVector)
         n = length(v)
         if n == 0; return result; end
-        ($fp)(v, result, $(op==:+ ? :(zero(first(v))) : :(one(first(v)))), first(indices(v,1)), n)
+        li = linearindices(v)
```

Comment:
Doesn't `length(v)` return the number of elements in the array regardless?  Or is `length(v)` equivalent to `last(linearindices(v))`?  The latter would seem odd to me.